### PR TITLE
some spaces formatting

### PR DESCRIPTION
**Pull request by a14n**
The rules applied in this PR are:

* formatting `if` statements to have :
  * 1 space between `if` and the condition,
  * 1 space between the condition and the left bracket,
  * 1 space around `else` (if brackets)
  * no space between the condition and parenthesis : `if (condition) {`
* formatting `for` and `while` to have similar rules as if above.
* uniform indentation of initializer lists in constructors. To avoid to many changes, I choosed the most used format. In the future we could change it to always put the initializer list on column 9 (what dartfmt do)
* formatting maps:
  * no spaces next to brackets for one liners.
  * no space between key and `:`
  * 1 space between `:` and value

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/date_picker.dart [hunk 1]
```diff
@@ -1181,8 +1178,7 @@ class _CupertinoTimerPickerState extends State<CupertinoTimerPicker> {
               ),
             ),
           );
-        }
-        else
+        } else
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/date_picker.dart]: this one should have an opening curly too

File: packages/flutter/lib/src/cupertino/tab_view.dart [hunk 2]
```diff
@@ -165,8 +165,7 @@ class _CupertinoTabViewState extends State<CupertinoTabView> {
     if (name == Navigator.defaultRouteName && widget.builder != null) {
       routeBuilder = widget.builder;
       title = widget.defaultTitle;
-    }
-    else if (widget.routes != null)
+    } else if (widget.routes != null)
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/tab_view.dart]: same here. If if has curlies, else if should too

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 3]
```diff
@@ -1701,8 +1701,7 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
     if (widget._labelShouldWithdraw != old._labelShouldWithdraw && widget.decoration.hasFloatingPlaceholder) {
       if (widget._labelShouldWithdraw) {
         _floatingLabelController.forward();
-      }
-      else
+      } else
```
Comment (xster) [inline packages/flutter/lib/src/material/input_decorator.dart]: ditto

File: packages/flutter/lib/src/foundation/binding.dart [hunk 4]
```diff
@@ -323,7 +323,7 @@ abstract class BindingBase {
           await setter(parameters['enabled'] == 'true');
           _postExtensionStateChangedEvent(name, await getter() ? 'true' : 'false');
         }
-        return <String, dynamic>{ 'enabled': await getter() ? 'true' : 'false' };
+        return <String, dynamic>{'enabled': await getter() ? 'true' : 'false'};
```
Comment (xster) [inline packages/flutter/lib/src/foundation/binding.dart]: seems like we're not even consistent in our style guide about this 

between https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#dartdoc-specific-requirements and the header right below it

File: packages/flutter/lib/src/cupertino/date_picker.dart (hunk 1 above)
Comment (a14n) [reply packages/flutter/lib/src/cupertino/date_picker.dart]: done

File: packages/flutter/lib/src/foundation/binding.dart (hunk 4 above)
Comment (a14n) [reply packages/flutter/lib/src/foundation/binding.dart]: I'm not sure to understand what you mean.

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 5]
```diff
@@ -1012,12 +1012,10 @@ class _NavigationBarStaticComponents {
 
     if (userLeading != null) {
       leadingContent = userLeading;
-    } else if (
-      automaticallyImplyLeading &&
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: This seems fine. I'm not sure we need to tool-enforce this.

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 6]
```diff
@@ -1052,13 +1050,11 @@ class _NavigationBarStaticComponents {
     @required ModalRoute<dynamic> route,
     @required bool automaticallyImplyLeading,
   }) {
-    if (
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: ditto here and in general

File: packages/flutter/lib/src/rendering/table.dart [hunk 7]
```diff
@@ -881,11 +881,11 @@ class RenderTable extends RenderBox {
         }
         assert(tableWidth >= targetWidth);
       }
-    } else // step 2 and 3 are mutually exclusive
+    } // step 2 and 3 are mutually exclusive
 
     // 3. if there were no flex columns, then grow the table to the
     //    minWidth.
-    if (tableWidth < minWidthConstraint) {
+    else if (tableWidth < minWidthConstraint) {
```
Comment (xster) [inline packages/flutter/lib/src/rendering/table.dart]: I'd move the comments around a bit but I wouldn't leave the else if to start on its own line.

**Review by xster — COMMENTED** [review]
I wouldn't change the 
```dart
if (
  A &&
  B &&
  C
) {
```
stuff, but the rest LGTM.

File: packages/flutter/lib/src/rendering/table.dart (hunk 7 above)
Comment (a14n) [reply packages/flutter/lib/src/rendering/table.dart]: We agree on the fact that `else if` should never be splitted.

I allowed this pattern for several reasons:

```dart
    var m;
    if (m) {
      m;
    }
    // my comment on this else branch
    else if (m) {
      m;
    }
    // my comment on this else branch
    else if (m) {
      m;
    }
```

looks better IMHO than end of line comments:

```dart
    var m;
    if (m) {
      m;
    } else if (m) { // my comment on this else branch that can not be multiline and should be short to not goes to far at right
      m;
    } else if (m) { // my comment on this else branch
      m;
    }
```

or

```dart
    var m;
    if (m) {
      m;
    } else if (m) {
      // my comment on this else branch here seems to reference the code that follows this comment and not the condition.
      m;
    } else if (m) {
      // my comment on this else branch
      m;
    }
```

Finally, `else if` may already start its own line without blocks:

```dart
    var m;
    if (m)
      m;
    else if (m)
      m;
```

**Review by a14n — COMMENTED** [review]
I change the rule to allow multiline condition in if to be on its own lines.

**Pull request merged**